integration-tests/test/helpers/testHelpers.js

```diff
@@ -2,31 +2,39 @@ import connectionHelper from "../lib/JavabuilderConnectionHelper.js";
 import {expect} from "chai";
 
 /**
- * Helper for verifying the basic case that set of messages was received
+ * Helper for verifying that a set of messages was received (and possibly sent)
  * from Javabuilder. Fails if there are any errors connecting.
  *
  * @param {*} sourcesJson sources (see sources.js)
  * @param {*} miniAppType mini-app type (see MiniAppType.js)
- * @param {*} assertOnMessagesReceived a verification callback to be called once all messages have been received
+ * @param {*} assertOnMessagesObserved a verification callback to be called once all messages have been sent/received
  * @param {*} doneCallback Mocha's 'done' callback
+ * @param {*} onMessageCallback a callback executed after each received message that can be used to respond to messages
```

Comment:
I'm curious, do you think it would just be easier to not use this test helper for interactive programs and instead call functions on `JavabuilderConnectionHelper` directly? We could expose a `sendMessage()` that could be called directly, so you wouldn't have to pass an extra callback in here. Just a thought - this approach also seems to work fine

Reply:
Totally missed this message -- your comment is kind of what I ended up having to do to test having a bad token in my [error cases PR](https://github.com/code-dot-org/javabuilder/pull/312/files), and I've been feeling like I'm piping callbacks around too much generally. So this might be just what I need to do, I'll look into it thanks!

Reply:
Looking at it a bit, I do feel like I still need a lot of the stuff that `verifyMessages` provides (collect all messages in `onMessage`, verify messages in `onClose`), which I think I'd need to replicate or move to call `JavabuilderConnecionHelper` directly. Maybe I'll just leave as-is?